src/main/java/org/opensearch/knn/index/engine/MethodComponentContext.java

```diff
@@ -49,6 +49,29 @@ public class MethodComponentContext implements ToXContentFragment, Writeable {
     private final String name;
     private final Map<String, Object> parameters;
 
+    /**
+     * Copy constructor. Creates a deep copy of a {@link MethodComponentContext}
+     *
+     * @param methodComponentContext to be copied. Must NOT be null
+     */
+    public MethodComponentContext(MethodComponentContext methodComponentContext) {
```

Comment:
Can we rely on lombok builders instead? you can do` @Builder(toBuilder = true)` and then `methodcomponentContext.toBuilder().build()` to create copy. Managing copying this way is error prone and forces us to maintain it

Reply:
Does `methodcomponentContext.toBuilder().build()` create a deep copy? Reason I ask is that there is a recursive component and I dont want to risk doing shallow copies.

Reply:
Probably not, this works for now